### PR TITLE
Stats: Check if Jetpack is integrated with the Complianz plugin to show the notice

### PR DESCRIPTION
**Pull request by dognose24**
Related to https://github.com/Automattic/jetpack-roadmap/issues/1517.

## Proposed changes:

* Check if the `Complianz` plugin is installed by WPCOM API.
* Check if Jetpack is integrated with the `Complianz` plugin by getting the option `complianz_options_integrations`.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

N/A

## Testing instructions:

* Install the [plugin Complianz](https://wordpress.com/plugins/complianz-gdpr) and complete the settings to launch the cookie consent banner.

截圖 2024-06-14 下午3 01 56

* Follow the Odyssey Stats testing instructions at https://github.com/Automattic/wp-calypso/pull/91762.
* Ensure the GDPR cookie consent notice shows with the plugin page link working correctly.
* Disable the plugin integration with Jetpack.

截圖 2024-06-14 下午3 00 43

* Ensure the GDPR cookie consent notice does not appear.

### CODE REVIEW

File: projects/packages/stats-admin/src/class-notices.php [hunk 1]
```diff
@@ -87,7 +92,8 @@ public function get_notices_to_show() {
 				self::OPT_OUT_NEW_STATS_NOTICE_ID  => $new_stats_enabled
 					&& $stats_views < self::VIEWS_TO_SHOW_FEEDBACK
 					&& ! $this->is_notice_hidden( self::OPT_OUT_NEW_STATS_NOTICE_ID ),
-			)
+			),
+			array( self::GDPR_COOKIE_CONSENT_NOTICE_ID => $is_jetpack_blocked_by_gdpr )
```
Comment (kangzj) [inline projects/packages/stats-admin/src/class-notices.php]:
```
&& ! $this->is_notice_hidden( self::OPT_OUT_NEW_STATS_NOTICE_ID )
```

File: projects/packages/stats-admin/src/class-notices.php [hunk 2]
```diff
@@ -70,23 +71,33 @@ public function get_notices_to_show() {
 		$stats_views              = intval( Stats_Options::get_option( 'views' ) );
 		$odyssey_stats_changed_at = intval( Stats_Options::get_option( 'odyssey_stats_changed_at' ) );
 
+		// Check if Jetpack is integrated with the Complianz plugin, which blocks the Stats.
+		$complianz_options_integrations  = get_option( 'complianz_options_integrations' );
+		$is_jetpack_blocked_by_complianz = ! isset( $complianz_options_integrations['jetpack'] ) || $complianz_options_integrations['jetpack'];
+		$is_complianz_installed          = $notices_wpcom[ self::GDPR_COOKIE_CONSENT_NOTICE_ID ];
```
Comment (kangzj) [inline projects/packages/stats-admin/src/class-notices.php]: ah.. I guess we could remote this line now right?

Comment (kangzj) [reply projects/packages/stats-admin/src/class-notices.php]: addressed in adc247a977

**Review by kangzj — DISMISSED** [review]
LGTM!

File: projects/packages/stats-admin/src/class-notices.php (hunk 2 above)
Comment (dognose24) [reply projects/packages/stats-admin/src/class-notices.php]: I updated the local installation check to work compatibly with API without the key `GDPR_COOKIE_CONSENT_NOTICE_ID` in [3ea6d10](https://github.com/Automattic/jetpack/pull/37870/commits/3ea6d10c2d5b8741f7a03560e1ff2e72d9303a33).

**Review by kangzj — DISMISSED** [review]

**Review by kangzj — DISMISSED** [review]

**Review by kangzj — APPROVED** [review]

**Pull request merged**